src/charm.py

```diff
@@ -892,22 +905,19 @@ def _is_cpu_compatible(self) -> bool:
             required_extension in self._get_cpu_extensions()
             for required_extension in REQUIRED_CPU_EXTENSIONS
         ):
-            self.unit.status = BlockedStatus(
+            return BlockedStatus(
```

Comment:
This seems like a one hell of a long status message to be printed in a Juju status screen. Does it even fit the table nicely?
How about we keep the return type of the whole method as a `bool`, log the error message rather than print it in the status, set the `Blocked` status in the `_on_collect_unit_status` if this returns `False` and add a generic status message?
WDYT?

Reply:
BTW, if we go with the current impl, docstring needs to be updated.

Reply:
What about returning only two different messages (one for CPU not compatible, one for not HugePages available)? Otherwise we lose information on whether the blocked status depends on CPU or HugePages.

Reply:
We don't lose the information - what I was thinking was logging the exact reason:
```python
logger.error("The exact error desc goes here")
return False
```
And then in the `_unit_is_non_active_status`, something along the lines of:
```python
if not self._is_cpu_compatible():
    return BlockedStatus("Incompatible CPU. Check logs for details.")
```